enterprise/backup/src/test/java/org/neo4j/backup/BackupToolCmdArgumentsAcceptanceTest.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.backup;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.io.PrintStream;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import java.util.Map;
+
+import org.neo4j.helpers.collection.Iterables;
+import org.neo4j.kernel.configuration.Config;
+
+import static java.util.Arrays.asList;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.neo4j.helpers.collection.MapUtil.stringMap;
+
+/**
+ * This test builds all valid combinations/permutations of args for {@link org.neo4j.backup.BackupTool} and asserts
+ * that it can handle those.
+ * It tests legacy and modern sets of args in all possible forms: (-option, --option, -option value, -option=value).
+ * Legacy is (-from, -to, -verify) and modern is (-host, -port, -to, -verify).
+ */
+
+@RunWith( Parameterized.class )
+public class BackupToolCmdArgumentsAcceptanceTest
+{
+    private static final String HOST = "localhost";
+    private static final int PORT = 9090;
+    private static final String PATH = "/var/backup/neo4j/";
+
+    @Parameter( 0 )
+    public String argsAsString;
+    @Parameter( 1 )
+    public boolean expectedVerifyStoreValue;
+
+    @Parameters( name = "args=({0})" )
```

Comment:
@davidegrohmann turned out there exist no format for printing 2D arrays via `MessageFormat`. that is why i've made first parameter `String` where all args are just joined with space as separator. Test method splits this string in line 88. Does this look sane?